### PR TITLE
[remark-lint] Use `.git` directory and add `remark-validate-links` to default ruleset

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

For example, The plugin `remark-validate-links` scans a `.git/` directory.
So, I think it useful to use `.git` during analysis.

Note:
- `remark` finds only files with Markdown extensions, so there are no problems even if `.git` exists.
- This adds `remark-validate-links` to the pre-installed package list.

See also:
- https://github.com/remarkjs/remark-validate-links/blob/10.0.4/readme.md#configuration
- https://github.com/sider/sider-docs/pull/584

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

None.

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

**Comment by ybiquitous** [comment]
@yykamei Could you please review this PR if you have time?

**Review by yykamei — COMMENTED** [review]
Can you include a smoke test with a `.git` directory?

And, I'd like to know the behavior with `root_dir`. I guess Sider's config, `root_dir`, does not affect the remark-lint analyses, but I'm curious whether a relative link is also successfully scanned :pray:

**Comment by ybiquitous** [comment]
> Can you include a smoke test with a `.git` directory?

The smoke test case is added already, but is it sufficient?
When `#use_git_metadata_dir?` return `false`, the following `use_git` case fails:

```console
$ bundle exec rake docker:build docker:smoke ANALYER=remark_lint ONLY=use_git DEBUG=trace
...
{ :trace => "error", :message => "Command failed: git remote -v\nfatal: not a git repository (or any of the parent directories): .git\n (at `foo.md`; rule: <none>)\nError: Command failed: git remote -v\nfatal: not a git repository (or any of the parent directories): .git\n\n    at ChildProcess.exithandler (child_process.js:308:12)\n    at ChildProcess.emit (events.js:315:20)\n    at maybeClose (internal/child_process.js:1048:16)\n    at Socket.<anonymous> (internal/child_process.js:439:11)\n    at Socket.emit (events.js:315:20)\n    at Pipe.<anonymous> (net.js:673:12)", :recorded_at => "2021-03-24T09:10:53.234Z", :truncated => false }
...
❌ Pattern matching failed at [:result][:message]:
--- expected
+++ actual
@@ -1 +1 @@
-nil
+"1 error(s) reported. See the log for details."
❌ use_git (3.9 seconds)

❌ 0 passed, 1 failed, 1 total (3.9 seconds)
```

> And, I'd like to know the behavior with `root_dir`. I guess Sider's config, `root_dir`, does not affect the remark-lint analyses, but I'm curious whether a relative link is also successfully scanned 🙏

Good catch! I will add a case. 👍

**Comment by ybiquitous** [comment]
> Good catch! I will add a case. 👍

Done. cfc963b

**Comment by ybiquitous** [comment]
It seems useful to add `remark-validate-links` to the default ruleset. 🤔

**Comment by yykamei** [comment]
> Done. [cfc963b](https://github.com/sider/runners/commit/cfc963b6929196b528146679709409c7563acdd9)

Thank you!

> The smoke test case is added already, but is it sufficient?
> When `#use_git_metadata_dir?` return `false`, the following `use_git` case fails:

I wondered if `#add_test_with_git_metadata` should be used in smoke tests just in case. Of course, if it's unnecessary, ignore my comment :smile:

**Comment by ybiquitous** [comment]
> It seems useful to add `remark-validate-links` to the default ruleset. 🤔

I tried it via 248d2026.

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ Misc:
 - **TSLint** Extend deadline and add helpful link [#2215](https://github.com/sider/runners/pull/2215)
 - Do not use tmpdir for working directory [#2217](https://github.com/sider/runners/pull/2217)
 - `Runners::Config` check warnings [#2221](https://github.com/sider/runners/pull/2221)
+- **remark-lint** Use `.git` directory [#2224](https://github.com/sider/runners/pull/2224)
```
Comment (ybiquitous) [inline CHANGELOG.md:39]:
```suggestion
- **remark-lint** Use `.git` directory and add `remark-validate-links` to default ruleset [#2224](https://github.com/sider/runners/pull/2224)
```

**Comment by ybiquitous** [comment]
I've finished the modification. Could you review it again? @yykamei

**Review by yykamei — APPROVED** [review]
LGTM

**Comment by ybiquitous** [comment]
Thank you 😃

**Pull request merged**